lxd/networks.go

```diff
@@ -1845,6 +1846,26 @@ func networkStateGet(d *Daemon, r *http.Request) response.Response {
 		return response.SmartError(api.StatusErrorf(http.StatusNotFound, "Network not found"))
 	}
 
+	// If network is of type physical managed, then return info on parent.
+	if n != nil && n.Type() == "physical" && n.IsManaged() {
+		config := n.Config()
+		state, err := resources.GetNetworkState(network.GetHostDevice(config["parent"], config["vlan"]))
+		if err != nil {
+			// If parent is not found, then return response indicating the parent is unavailable.
+			statusErr := api.StatusError{}
+			if errors.As(err, &statusErr) && statusErr.Status() == 404 {
+				return response.SyncResponse(true, &api.NetworkState{
+					State: "unavailable",
+					Type:  "unknown",
+				})
+			}
+
+			return response.SmartError(err)
+		}
+
+		return response.SyncResponse(true, state)
+	}
+
 	var state *api.NetworkState
 	if n != nil {
 		state, err = n.State()
```

Comment:
I've just seen this part of the code now (we didn't get this far down in our meeting) and I see now that the Network interface has a `State()` function that allows delegating the generation of the network state info to each Network driver.

So we shouldn't be adding network driver specific logic into this `networkStateGet` function but instead pushing down the driver specific implementation into the driver.

So we should be creating a State() function in https://github.com/canonical/lxd/blob/main/lxd/network/driver_physical.go to override the common function.

Reply:
Moved the logic to the driver. Will remember to do so for similar tasks in the future. Thanks!